syntax = "proto3";

package protos;



import "google/protobuf/empty.proto";
import "google/protobuf/timestamp.proto";
import "custom_types.proto";
//TODO
service LoginLogService {
  rpc GetAll (GetAllRequest) returns (GetAllLoginLogsResponse);
  rpc GetById (GetLoginLogByIdRequest) returns (LoginLogModel);
  rpc GetByCustomer (GetLoginLogsByCustomerRequest) returns (GetAllLoginLogsResponse);
  rpc GetInTimeRange (GetLoginLogsInTimeRangeRequest) returns (GetAllLoginLogsResponse);
}

message LoginLogModel {
  int32 log_id = 1;
  optional int32 customer_id = 2;
  optional google.protobuf.Timestamp login_time = 3;
  optional string ip_address = 4;
  optional string device_info = 5;
}

message GetAllLoginLogsResponse {
  repeated LoginLogModel login_logs = 1;
}

message GetLoginLogByIdRequest {
  int32 log_id = 1;
}

message GetLoginLogsByCustomerRequest {
  int32 customer_id = 1;
}

message GetLoginLogsInTimeRangeRequest {
  google.protobuf.Timestamp start_time = 1;
  google.protobuf.Timestamp end_time = 2;
}
